packages/payment-proxy-client/src/file.ts

```diff
@@ -38,3 +33,91 @@ function parseFileNameFromUrl(url: string): string {
     return "fetched-file";
   }
 }
+
+export async function fetchFileInChunks(
+  chunkSize: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient
+): Promise<File> {
+  const firstChunk = await fetchFileChunk(
+    0,
+    chunkSize - 1,
+    baseUrl,
+    costPerByte,
+    selectedChannel,
+    nitroClient
+  );
+
+  const { contentLength, fileName } = firstChunk;
+  const remainingChunks = Math.ceil(contentLength / chunkSize) - 1;
+
+  const fileContents = new Uint8Array(contentLength);
+  fileContents.set(firstChunk.data);
+
+  for (let i = 1; i <= remainingChunks; i++) {
+    const start = i * chunkSize;
+    const stop = start + chunkSize - 1;
+
+    const { data } = await fetchFileChunk(
+      start,
+      stop,
+      baseUrl,
+      costPerByte,
+      selectedChannel,
+      nitroClient
+    );
+
+    fileContents.set(data, i * chunkSize);
+  }
+
+  return new File([fileContents], fileName);
+}
+
+export async function fetchFileChunk(
+  start: number,
+  stop: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient
+): Promise<{ data: Uint8Array; contentLength: number; fileName: string }> {
+  const dataLength = stop - start + 1; // +1 because stop is inclusive
+  console.log(dataLength);
+  const chunkCost = dataLength * costPerByte;
+
+  const voucher = await nitroClient.CreateVoucher(selectedChannel, chunkCost);
+
+  const req = new Request(
+    `${baseUrl}?channelId=${voucher.ChannelId}&amount=${voucher.Amount}&signature=${voucher.Signature}`
+  );
+  req.headers.set("Range", `bytes=${start}-${stop}`);
+
+  const response = await fetch(req);
+  if (!response.body) {
+    throw new Error("Response body is null");
+  }
+
+  if (!response.ok) {
+    throw new Error(`Response status ${response.status}`);
+  }
+  const result = await response.body.getReader().read();
+  console.log(...response.headers);
+
+  return {
+    data: result.value || new Uint8Array(),
+    contentLength: parseTotalSizeFromContentRange(
+      response.headers.get("Content-Range") || ""
+    ),
+    fileName: parseFileNameFromUrl(response.url),
+  };
+}
+
+function parseTotalSizeFromContentRange(contentRange: string): number {
```

Comment:
Should we make a unit test for this?

Reply:
Right now this is just part of the payment proxy demo, which lacks tests completely, so I think it's OK for now to leave it without a test.